panels/user-accounts/cc-app-permissions.c

```diff
@@ -191,26 +192,54 @@ reload_apps (CcAppPermissions *self)
           app_name[0] == '\0' ||
           /* Endless' link apps have the "eos-link" prefix, and should be ignored too */
           g_str_has_prefix (g_app_info_get_id (app), "eos-link") ||
-          /* FIXME: Only list flatpak apps for now; we really need a system-wide MAC
-           * to be able to reliably support blacklisting system programs. See
+          /* FIXME: Only list flatpak apps and apps with X-Parental-Controls
+           * key set for now; we really need a system-wide MAC to be able to
+           * reliably support blacklisting system programs. See
            * https://phabricator.endlessm.com/T25080. */
-          !g_desktop_app_info_has_key (G_DESKTOP_APP_INFO (app), "X-Flatpak"))
+          (!g_desktop_app_info_has_key (G_DESKTOP_APP_INFO (app), "X-Flatpak") &&
+           !g_desktop_app_info_has_key (G_DESKTOP_APP_INFO (app), "X-Parental-Controls")))
         {
           continue;
         }
 
-      flatpak_id = g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (app), "X-Flatpak");
-      g_debug ("Processing app ‘%s’ (Exec=%s, X-Flatpak=%s)",
-               g_app_info_get_id (app),
-               g_app_info_get_executable (app),
-               flatpak_id);
+      if (g_desktop_app_info_has_key (G_DESKTOP_APP_INFO (app), "X-Flatpak"))
+        {
+          g_autofree gchar *flatpak_id = NULL;
+
+          flatpak_id = g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (app), "X-Flatpak");
+          g_debug ("Processing app ‘%s’ (Exec=%s, X-Flatpak=%s)",
+                   g_app_info_get_id (app),
+                   g_app_info_get_executable (app),
+                   flatpak_id);
 
-      /* Have we seen this flatpak ID before? */
-      if (!g_hash_table_add (seen_flatpak_ids, g_steal_pointer (&flatpak_id)))
+          /* Have we seen this flatpak ID before? */
+          if (!g_hash_table_add (seen_flatpak_ids, g_steal_pointer (&flatpak_id)))
+            {
+              g_debug (" → Skipping ‘%s’ due to seeing its flatpak ID already",
+                       g_app_info_get_id (app));
+              continue;
+            }
+        }
+      else if (g_desktop_app_info_has_key (G_DESKTOP_APP_INFO (app), "X-Parental-Controls"))
         {
-          g_debug (" → Skipping ‘%s’ due to seeing its flatpak ID already",
-                   g_app_info_get_id (app));
-          continue;
+          g_autofree gchar *parental_controls_type = NULL;
+          g_autofree gchar *executable = NULL;
+
+          parental_controls_type = g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (app),
```

Comment:
For future-proofness, maybe ignore desktop files which have `X-Parental-Controls=none`?

Reply:
Done.